plugins/ngrok/provisioner.go

```diff
@@ -0,0 +1,81 @@
+package ngrok
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	version           = "2"
+	apiKeyYamlName    = "api_key"
+	authTokenYamlName = "authtoken"
+	versionYamlName   = "version"
+)
+
+type fileProvisioner struct {
+}
+
+func newNgrokProvisioner() sdk.Provisioner {
+	return fileProvisioner{}
+}
+
+func (f fileProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	provisionedConfigFilePath := filepath.Join(in.TempDir, "config.yml")
+	config := make(map[string]interface{})
+	configFilePath := processConfigFlag(out, provisionedConfigFilePath)
+	if configFilePath != "" {
+		existingContents, err := os.ReadFile(configFilePath)
+		if err != nil {
+			out.AddError(err)
+			return
+		}
+
+		if err := importer.FileContents(existingContents).ToYAML(&config); err != nil {
+			out.AddError(err)
+			return
+		}
+	}
+
+	config[authTokenYamlName] = in.ItemFields[fieldname.Authtoken]
+	config[apiKeyYamlName] = in.ItemFields[fieldname.APIKey]
+	config[versionYamlName] = version
+
+	newContents, err := yaml.Marshal(&config)
+	if err != nil {
+		out.AddError(err)
+		return
+	}
+
+	out.AddSecretFile(provisionedConfigFilePath, newContents)
+}
+
+func processConfigFlag(out *sdk.ProvisionOutput, newFilePath string) string {
```

Comment:
This function name doesn't really reflect what it does. Let's consider changing it to reflect that it replaces the path passed by the user with a temp path and returns the initial path, and add a comment to it.

Reply:
Done, thanks!